### ISSUE TITLE
Fine tuning RetinaNet models

### ISSUE THREAD

**Issue by Paryavi** [keras-team/keras-cv#2055]
**Short Description**

**Papers**
 Relevant Link: https://blog.research.google/2020/04/efficientdet-towards-scalable-and.html

**Existing Implementations**

[**Other Information**] Also, if possible, adding capability for developer to add their own model eventually)

**Comment by Paryavi** [comment]
Hi @ianstenbit,
Also, do you have a KerasCV example explaining how to fine-tune the models(RetinaNEt, YoLo), like modifying parameters (mostly the layers or filter dimension) that we use for fine-tuning, similar to what we have for image classification fine-tuning?

**Comment by jbischof** [comment]
@Paryavi have you checked out https://keras.io/guides/keras_cv/object_detection_keras_cv/?

**Comment by Paryavi** [comment]
Yes, I think for filtration I found a relevant example;

backbone = keras_cv.models.ResNetBackbone(
    stackwise_filters=[64, 128, 256, 512],
    stackwise_blocks=[2, 2, 2, 2],
    stackwise_strides=[1, 2, 2, 2],
    include_rescaling=False,
)
Source; https://keras.io/api/keras_cv/models/
But for finetuning the exact layer and freezing a layer, I am still not sure how it is done in KerasCV, I mean in classification we use something like;  
https://github.com/keras-team/keras-io/blob/7ef56500e88cdd3323abece248450c3991905199/examples/vision/image_classification_efficientnet_fine_tuning.py#L357

**Comment by Paryavi** [comment]
Sorry, this link shows layer selection in classification, not sure how to do the same in detection; 
https://github.com/keras-team/keras-io/blob/7ef56500e88cdd3323abece248450c3991905199/examples/vision/image_classification_efficientnet_fine_tuning.py#L424

**Comment by jbischof** [comment]
@Paryavi KerasCV is just an extension of Keras! You can use the same approach as your linked example.

If you're looking for specific recommendations of layers to freeze, etc, I do understand the confusion. We ended up not freezing the batchnorm layers when training our OD guide and didn't get a meaningfully different result. There is a lot of application-specific experimentation required.

**Comment by jbischof** [comment]
@Paryavi I've changed the title of this issue. Feel free to file another for EfficientDet. Ideally we want one deliverable per issue 😄

**Comment by Paryavi** [comment]
Thanks @jbischof :)
Sure, will create a new thread now.
